Assets/Scripts/Drone/Tutorial/Tutorial_Movement02_DroneInterceptPlayer.cs

```diff
@@ -0,0 +1,145 @@
+    [Tooltip("After the cutscene has ended, this is the delay between when the drone starts moving toward the player and when the camera starts to turn to look at the drone.")]
+    [SerializeField, Min(0)]
+    private float _cameraLookAtDroneDelay = 1f;
+    [Tooltip("This sets how long it takes the drone to perform medical assistance when it intercepts the player right after the cutscene plays.")]
+    [SerializeField, Min(0)]
+    private float _medicalAssistanceDuration = 3f;
+
+    [Header("Drone Pictogram")]
+    [SerializeField]
+    private PictogramBehavior _pictogramBehaviour;
+    [SerializeField]
+    private Sprite _sprite_DroneMedicalAssistance;
+
+    [Header("Finite State Machine Parameters")]
+    [Tooltip("This parameter tracks what step the tutorial is currently in.")]
+    [SerializeField]
+    private FSMParameter _tutorialProgressParameter;
+
+    [Header("Tutorial State Machine Events - Stage 02 - Drone Intercepts Player")]
+    [SerializeField]
+    private GameEventScriptableObject _Enter;
+    [SerializeField]
+    private GameEventScriptableObject _Update;
+    [SerializeField]
+    private GameEventScriptableObject _Exit;
+
+
+
+    private FiniteStateMachineInstance _stateMachineInstance;
+
+    private Transform _player;
+
+
+    private GameEventResponses _gameEventResponses = new();
+
+    private bool _startedMedicalAssistance;
+    private bool _medicalAssistanceComplete;
+
+
+
+    private void Awake()
+    {
+        _stateMachineInstance = _drone.GetStateMachineInstance();
+
+        _player = Player.Motion.PlayerMovement.Instance.transform;
+
+        _gameEventResponses.SetResponses(
+            (_Enter, EnterTutorial),
+            (_Update, UpdateTutorial),
+            (_Exit, ExitTutorial)
+            );
+    }
+
+    private void OnEnable() => _gameEventResponses.Register();
+    private void OnDisable() => _gameEventResponses.Unregister();
+
+
+    private void EnterTutorial()
+    {
+        // Switch to the tutorial camera.
+        _tutorialCamera.MoveToTopOfPrioritySubqueue();
+
+        _medicalAssistanceComplete = false;
+        _startedMedicalAssistance = false;
+
+        _pictogramBehaviour.ChangePictogramImage(_sprite_DroneMedicalAssistance);
+
+        StartCoroutine(WaitToFocusCamera());
+    }
+
+    private void UpdateTutorial()
+    {
+        // Make the drone move into view until it is right in front of the player.
+        if (!_startedMedicalAssistance &&
+            _movement.MoveDroneInFrontOfPlayer(_droneDistanceToStopInFrontOfPlayer, true))
+        {
+            // The drone has reached the player, so display the health image.
+            _pictogramBehaviour.SetImageActive(true);
+
+            _startedMedicalAssistance = true;
+
+            StartCoroutine(WaitForMedicalAssistanceToFinish());
+        }
+
+
+        if (_medicalAssistanceComplete)
+        {
+            // Trigger transition to next part of the tutorial.
```

Comment:
Maybe you could just use a trigger parameter to advance to the next step. That way it's easier to make changes like inserting another step.